tools/logger/convert.c

```diff
@@ -640,10 +649,13 @@ static int logger_read(void)
 					return -errno;
 				}
 			}
-			log_err("in %s(), fread(..., %s) failed: %s(%d)\n",
-				 __func__, global_config->in_file,
-				strerror(errno), errno);
-			return -errno;
+			if (!feof(global_config->in_fd)) {
+				log_err("in %s(), fread(..., %s) failed: %s(%d)\n",
+					__func__, global_config->in_file,
```

Comment:
Sorry I removed `ferror()` here in commit 337afc4f3809, my bad. After doing my homework (better late than never) I think it's better to add `ferror()` back here and use it instead of `!feof()` because the standard doesn't state that `feof` and `ferror` are mutually exclusive. In fact you could imagine both of them being true.

https://pubs.opengroup.org/onlinepubs/9699919799/functions/fread.html

Reply:
BTW the `if(!ret)` is really confusing, please change it to `if(ret<1)` to make it consistent with the `fread()` specification.

Reply:
Updated